pkg/diff/diff.go

```diff
@@ -298,13 +299,11 @@ func (t *TableDiff) checkChunkDataEqual(ctx context.Context, checkJobs []*CheckJ
 	return equal, nil
 }
 
-func (t *TableDiff) compareRows(sourceRows map[string]*sql.Rows, targetRows *sql.Rows, orderKeyCols []*model.ColumnInfo) (bool, error) {
+func (t *TableDiff) compareRows(sourceRows map[string][]map[string]*dbutil.ColumnData, targetRows []map[string]*dbutil.ColumnData, orderKeyCols []*model.ColumnInfo) (bool, error) {
```

Comment:
why change it？

Reply:
we should read data from sql.Rows and then close the rows.

Reply:
if use `rows`, we can read it while we need, now we should read all once, right?

but you can keep it, while you need to refine mem usage, you should change it back

Reply:
ok